### ISSUE TITLE
[BUG]: Channels doesn't sort alphabetically

### ISSUE THREAD

**Issue by r00tnpm** [yuliskov/SmartTube#4097]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/yuliskov/SmartTube/issues) or [closed](https://github.com/yuliskov/SmartTube/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://github.com/yuliskov/SmartTube#faq) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.

### Affected version

24.61

### Device Type

Smart TV/Box

### Affected Android

Shield TV

### Steps to reproduce the bug

1. Go to "Channels" section
2. See the channel sorting order

### Actual behavior

The channel section used to sort alphabetically. The channels with updated contents would be on top, followed by all other channels A~Z. However, with the latest version 24.61, it appears non-alphabetical, which made it very hard to track down channels if user has a lot of subscriptions ... please help fix the issue, thanks!

**Comment by r00tnpm** [comment]
Btw, my YT channel in chrome browser has alphabetical setting, all channels appear A~Z just fine, so it is not a user setting (YT) issue.

**Comment by Kouyate24** [comment]
Same for me too, channels in random order and when I select the option to do them alphabetically (fast, animated previews), the channel list is blank. PS just did latest update today so using version 24.61

**Comment by jkr4m3r** [comment]
Getting this on my Shield also, update on 14-Nov says it fixed it, but didn't.

**Comment by KjetilHelgesen** [comment]
Same for me as well on a Philips TV. Tried changing the channel sorting setting, seems the first two settings ("Default fast animated previews" and "Alphabetically fast animated previews" don't work at all, "Alphabetically" does not sort alphabetically, but random, "New content" seems to work OK, and "Last viewed" doesn't work.

**Comment by DiscoFever78** [comment]
Alphabetical sorting is such a core/basic programming element... How is this bug even a thing??

**Comment by Seaqull** [comment]
Also having  s problem with the shorts replaying instead of going back to channel list. You have to pause the video and use the back button.

**Comment by jkr4m3r** [comment]
Yea, I'm having lots of different errors now not related to this bug, so that tells me YouTube is changing stuff on their end and the poor guy(s) who code this great app have to sort it all out. Hate to be him/them right now. 

That reminds me, time to hit up the Patreon page...

**Comment by sweharris** [comment]
Looks like channel ordering backend has changed, at least according to this LTT forum post: https://linustechtips.com/topic/1589026-youtube-changed-channel-sort-order-functionality-on-tv-based-apps/

**Comment by edgan** [comment]
In `24.78` all the options work as expected, except `Alphabetically`. It behaves like `New Content`. Yet `Alphabetically(fast, animated previews)` is actually alphabetical.

**Comment by CJudders** [comment]
I've got 3 users on my Fire stick 4K Max 2023. I find whichever user is the last logged in before closing, that setting will be what you get for any user on next startup. My son has new content, while I have alphabetical. If he watches and I switch on later, navigate to my sub, my channels will be sorted by new content.

**Comment by Kouyate24** [comment]
There is a setting where you can set it each user has there own settings, have u tried changing that?

**Comment by CJudders** [comment]
Yes. I have mine set alphabetically, my son's have it to new content. They watch in the morning. When I watch in the evening after they're in bed, it's on one of their accounts. When I switch to mine, the channel sorting doesn't change with the switch. I have to load my profile and then restart smart tube to get the correct sorting.
